gen3/auth.py

```diff
@@ -378,3 +401,89 @@ def curl(self, path, request=None, data=None):
         else:
             raise Exception("Invalid request type: " + request)
         return output
+
+    def get_access_token_from_wts(self, endpoint=None):
+        """
+        Try to fetch an access token for the given idp from the wts
+        in the given namespace.  If idp is not set, then default to "local"
+        """
+        # attempt to get a token from the workspace-token-service
+        auth_url = get_wts_endpoint(self._wts_namespace) + "/token/"
+        if self._wts_idp and self._wts_idp != "local":
+            auth_url += "?idp={}".format(self._wts_idp)
+
+        try:
+            resp = requests.get(auth_url)
+            if (resp and resp.status_code == 200) or (not self._external_wts_host):
+                return _handle_access_token_response(resp, "token")
+        except Exception as e:
+            if not self._external_wts_host:
+                raise e
+            else:
+                logging.warning(
+                    "Accessing local WTS did not work. Trying network call since external WTS url is available"
+                )
+
+        # local workspace wts call failed, try using a network call
+        # First get access token with WTS host
+        wts_token = get_access_token_with_key(self._refresh_token)
+        auth_url = self._external_wts_host + "token/"
+
+        providerList = get_wts_idps(self._wts_namespace, self._external_wts_host)
```

Comment:
```suggestion
        provider_list = get_wts_idps(self._wts_namespace, self._external_wts_host)
```

python locally scoped variables should be `lowercase_underscore_separated`